connector-runtime/connector-runtime-core/src/main/java/io/camunda/connector/runtime/core/inbound/correlation/InboundCorrelationHandler.java

```diff
@@ -103,6 +110,70 @@ protected InboundConnectorResult<ProcessInstance> triggerStartEvent(
     }
   }
 
+  protected InboundConnectorResult<CorrelatedMessageStart> triggerMessageStartEvent(
+      InboundConnectorDefinitionImpl definition,
+      MessageStartEventCorrelationPoint correlationPoint,
+      Object variables) {
+
+    if (!isActivationConditionMet(definition, variables)) {
+      LOG.debug("Activation condition didn't match: {}", correlationPoint);
+      return new MessageStartCorrelationResult(
+          correlationPoint.messageName(),
+          new CorrelationErrorData(CorrelationErrorReason.ACTIVATION_CONDITION_NOT_MET));
+    }
+
+    String idempotencyKeyMessageId = extractIdempotencyKey(correlationPoint, variables);
+    if (correlationPoint.idempotencyKeyExpression() != null
+        && !correlationPoint.idempotencyKeyExpression().isBlank()
+        && idempotencyKeyMessageId == null) {
```

Comment:
Shouldnt `idempotencyKeyMessageId != null` check be enough here? I assume that `extractIdempotencyKey` already checks it?

Reply:
This is more like a safety check: when expression is defined but we won't able to extract it. Without this check we will return `500`.

Reply:
on its own, message id expression can be blank